lib/pure/times.nim

```diff
@@ -172,30 +178,43 @@ proc getDaysInYear*(year: int): int =
   ## Get the number of days in a ``year``
   result = 365 + (if isLeapYear(year): 1 else: 0)
 
-proc assertValidDate(monthday: MonthdayRange, month: Month, year: int) {.inline.} =
-  assert monthday <= getDaysInMonth(month, year),
-    $year & "-" & $ord(month) & "-" & $monthday & " is not a valid date"
+proc assertValidDate(d: Date) {.inline.} =
+  assert d.monthday <= getDaysInMonth(d.month, d.year),
+    $d.year & "-" & $ord(d.month) & "-" & $d.monthday & " is not a valid date"
+
+proc toEpochDate*(time: Time): EpochDate =
+  EpochDate(time.int64 div secondsInDay)
```

Comment:
I don't think this will be correct for times before 1970-01-01, e.g a time of `-1` will give the epoch date `0`, but it should be `-1`.

Reply:
fixed, should work now